app/src/main/java/org/commcare/dalvik/reminders/CommCareReceiver.kt

```diff
@@ -3,14 +3,24 @@ package org.commcare.dalvik.reminders;
 import android.content.BroadcastReceiver
 import android.content.Context
 import android.content.Intent
+import androidx.work.Data
 import androidx.work.OneTimeWorkRequest
 import androidx.work.WorkManager
 import org.commcare.dalvik.reminders.sync.SyncCasesWorker
 
 class CommCareReceiver : BroadcastReceiver() {
 
+    companion object {
+        const val CC_LOGGED_IN_USER_ID_KEY = "cc-logged-in-user-id"
+    }
     override fun onReceive(context: Context, intent: Intent) {
-        WorkManager.getInstance(context).enqueue(
-        OneTimeWorkRequest.from(SyncCasesWorker::class.java))
+        val data = Data.Builder()
+        if (intent.hasExtra(CC_LOGGED_IN_USER_ID_KEY)){
+            data.putString(CC_LOGGED_IN_USER_ID_KEY, intent.getStringExtra(CC_LOGGED_IN_USER_ID_KEY))
+        }
```

Comment:
`syncCasesWorkerRequest` is not always started by a broadcast. So havng the logged in user in the broadcast intent extra won't work. You will instead need to pull this data from commcare using content providers.

Reply:
@shubham1g5 what else triggers `syncCasesWorkerRequest`? This is to deal with the particular case in which a case update broadcast is sent before the case purge is completed, which causes the case (or cases) to-be-purged to be in the `resultset` of the query. So having the `CC_LOGGED_IN_USER_ID_KEY` `null` is kind of expected for most use cases. I wonder if this is applicable to the other components that may trigger `syncCasesWorkerRequest`.

Reply:
> what else triggers syncCasesWorkerRequest?

The Reminders app itself on startup triggers it without needing any broadcasts from CC.

I see what you mean. Confirming this works correctly - 

1. User creates a reminder case assigned to another user
2. Reminder Apps gets the broadcast and skips the case 
3. User closes and restarts Reminders App again 
4. Reminders app doesn't get the new case ?

Reply:
On point, the case would have been purged by then. The only scenario that this wouldn't be adequate is if the design of the app would make the `reminder` case stay in the issuer of the referral.

Reply:
I think the better change here is for CC to purge the cases before the broadcast.